### PR TITLE
Adding to_str to utils

### PR DESCRIPTION
**Pull request by RaphMota**

### CODE REVIEW

File: src/utils.jl [hunk 1]
```diff
@@ -1,3 +1,17 @@
...
+const MOI = MathOptInterface
+const MOIU = MOI.Utilities
+
+# indices
+const VI = MOI.VariableIndex
+const CI = MOI.ConstraintIndex
+
+# function aliases
+const SAF = MOI.ScalarAffineFunction{Float64}
+const SQF = MOI.ScalarQuadraticFunction{Float64}
```
Comment (joaquimg) [inline src/utils.jl:13]: remove

File: src/MOI/nlp.jl [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+const MOI = MathOptInterface
+const MOIU = MOI.Utilities
+
+# indices
+const VI = MOI.VariableIndex
+const CI = MOI.ConstraintIndex
+
+# function aliases
+const SAF = MOI.ScalarAffineFunction{Float64}
+const SQF = MOI.ScalarQuadraticFunction{Float64}
```
Comment (joaquimg) [inline src/MOI/nlp.jl:15]: remove

File: src/MOI/MOI_wrapper.jl [hunk 3]
```diff
@@ -132,6 +132,26 @@ mutable struct ConstraintInfo
...
 end
 
+mutable struct NLPVariableInfo
+    lower_bound::Union{Float64, Nothing}
+    upper_bound::Union{Float64, Nothing}
+    category::Symbol
+    start::Union{Float64, Nothing}
+    name::Union{String, Nothing}
+end
+NLPVariableInfo() = NLPVariableInfo(nothing, nothing, :Cont, nothing, nothing)
```
Comment (joaquimg) [inline src/MOI/MOI_wrapper.jl:142]: All this info already exists elsewhere. this should be removed as it is not necessary.

File: src/MOI/MOI_wrapper.jl [hunk 4]
```diff
@@ -211,7 +232,7 @@ mutable struct Optimizer <: MOI.AbstractOptimizer
         typeof(CleverDicts.key_to_index),
         typeof(CleverDicts.index_to_key),
         }
-
+    nlp_variable_info::Vector{NLPVariableInfo}
```
Comment (joaquimg) [inline src/MOI/MOI_wrapper.jl:235]: remove

File: src/MOI/MOI_wrapper.jl [hunk 5]
```diff
@@ -283,6 +307,8 @@ mutable struct Optimizer <: MOI.AbstractOptimizer
         model.solve_method = ""
         model.solve_relaxation = false
 
+        model.objective_expr = nothing
```
Comment (joaquimg) [inline src/MOI/MOI_wrapper.jl:310]: is this necessary?

File: src/MOI/MOI_wrapper.jl [hunk 6]
```diff
@@ -816,10 +849,13 @@ function MOI.add_variable(model::Optimizer)
         [-Inf],#_dbdl::Vector{Float64},
         [Inf],#_dbdu::Vector{Float64}
     )
+
+    push!(model.nlp_variable_info, NLPVariableInfo())
```
Comment (joaquimg) [inline src/MOI/MOI_wrapper.jl:853]: remove...

File: src/MOI/MOI_wrapper.jl [hunk 7]
```diff
@@ -816,10 +849,13 @@ function MOI.add_variable(model::Optimizer)
...
     return index
 end
 
 function MOI.add_variables(model::Optimizer, N::Int)
+
```
Comment (joaquimg) [inline src/MOI/MOI_wrapper.jl:858]: do not break lines in parts of the code you did no change

File: src/MOI/MOI_wrapper.jl [hunk 8]
```diff
@@ -906,19 +944,20 @@ end
...
+        check_variable_indices(model, v)
+        model.nlp_variable_info[v.value].name = name
+    else
+        info = _info(model, v)
+        info.name = name
+        # Note: don't set the string names in the Xpress C API because it complains
+        # on duplicate variables.
+        # That is, don't call `Lib.XPRSaddnames`.
+        model.name_to_variable = nothing
+    end
```
Comment (joaquimg) [inline src/MOI/MOI_wrapper.jl:957]: keep old code

File: src/MOI/MOI_wrapper.jl [hunk 9]
```diff
@@ -906,19 +944,20 @@ end
...
     return
 end
 
-###
-### Sensitivities
-###
```
Comment (joaquimg) [inline src/MOI/MOI_wrapper.jl:920]: do not remove

File: src/MOI/MOI_wrapper.jl [hunk 10]
```diff
@@ -1088,6 +1127,7 @@ function MOI.set(
     model::Optimizer, ::MOI.ObjectiveSense, sense::MOI.OptimizationSense
 )
     # TODO: should this propagate across a `MOI.empty!(optimizer)` call
+
```
Comment (joaquimg) [inline src/MOI/MOI_wrapper.jl:1130]: do not break lines in places you are not changing

File: src/utils.jl [hunk 11]
```diff
@@ -1,3 +1,17 @@
+import MathOptInterface
```
Comment (joaquimg) [inline src/utils.jl:1]: this file is necessary?

**Review by joaquimg — CHANGES_REQUESTED** [review]

File: src/MOI/MOI_wrapper.jl (hunk 6 above)
Comment (RaphMota) [reply src/MOI/MOI_wrapper.jl:853]: If I add another function MOI.add_variable with the same arguments for NLP case, MOI will always use the old function. Thus, it's the only solution I found to solve this issue

**Pull request closed without merging**